required/latex-lab/latex-lab-math.dtx

```diff
@@ -814,12 +837,23 @@
         \@@_grab_dollar:w
       }
   }
+
 \exp_args:No \tex_everydisplay:D
   {
     \tex_the:D \tex_everydisplay:D
     \iftrue  % this may have to be a settable flag!
 %      {
 %        \typeout{==>~ in~ everydisplay}
+%    \end{macrocode}
+% flipping the \cs{belowdisplay} values is done so that we get (assumption) 
```

Comment:
All the messing with these skips seems totally weird to me!  

Is there a better way? 

These changes clearly improve the situation.